source/javascripts/components/ConfigMergeDialog/ConfigMergeDialog.tsx

```diff
@@ -34,11 +47,11 @@ const readOnlyDiffEditorOptions: monaco.editor.IDiffEditorConstructionOptions =
   readOnly: true,
 };
 
-function mergeYamls(yourYaml: string, baseYaml: string, remoteYaml: string) {
+function mergeYamls(yourYaml: string, baseYaml: string, remoteYaml: string, finalYaml?: string) {
   const rows: string[] = [];
   const decorations: monaco.editor.IModelDeltaDecoration[] = [];
 
-  diff3Merge<string>(yourYaml, baseYaml, remoteYaml, { stringSeparator: '\n' }).forEach((region) => {
+  diff3Merge<string>(finalYaml || yourYaml, baseYaml, remoteYaml, { stringSeparator: '\n' }).forEach((region) => {
```

Comment:
even if the `finalYaml` exists isn't it the same always as `yourYaml`
`yourYaml` = `appService.appConfig`
`finalYaml` = `appService.appConfig` once set in the callback in the MainController